### ISSUE TITLE
error: clearing an object of non-trivial type; use assignment or value-initialization instead

### ISSUE THREAD

**Issue by imyxh** [RadeonOpenCompute/ROCR-Runtime#42]
I get this error while compiling from git.
```
/home/imyxh/git/ROCm/ROCR-Runtime/src/core/runtime/runtime.cpp: In member function ‘void core::Runtime::SetLinkCount(size_t)’:
/home/imyxh/git/ROCm/ROCR-Runtime/src/core/runtime/runtime.cpp:260:71: error: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘__gnu_cxx::__alloc_traits<std::allocator<core::Runtime::LinkInfo>, core::Runtime::LinkInfo>::value_type’ {aka ‘struct core::Runtime::LinkInfo’}; use assignment or value-initialization instead [-Werror=class-memaccess]
          link_matrix_.size() * sizeof(hsa_amd_memory_pool_link_info_t));
                                                                       ^
In file included from /home/imyxh/git/ROCm/ROCR-Runtime/src/core/runtime/runtime.cpp:43:
/home/imyxh/git/ROCm/ROCR-Runtime/src/core/inc/runtime.h:96:10: note: ‘__gnu_cxx::__alloc_traits<std::allocator<core::Runtime::LinkInfo>, core::Runtime::LinkInfo>::value_type’ {aka ‘struct core::Runtime::LinkInfo’} declared here
   struct LinkInfo {
          ^~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/hsa-runtime64.dir/build.make:349: CMakeFiles/hsa-runtime64.dir/core/runtime/runtime.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/hsa-runtime64.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

**Comment by jlgreathouse** [comment]
I don't. But it's probable that we're using different compiler versions. :)

What commands are you using to see this error? What distro? What version of gcc/g++ ?

**Comment by imyxh** [comment]
```
  gcc 8.2.1+20180831-1
linux 4.18.8.arch1-1
```
I get these errors when compiling with the instructions in the README, namely:
```
mkdir build
cd build
cmake -D CMAKE_PREFIX_PATH=/home/imyxh/git/ROCm/hsakmt ../src
make
```

**Comment by jlgreathouse** [comment]
I believe that this is fixed as of the release of the [ROCm 1.9.2 release](https://github.com/RadeonOpenCompute/ROCR-Runtime/releases/tag/roc-1.9.2) of ROCr. I can successfully build ROCr on Ubuntu 18.10 with GCC 8.2.0.
